calvin/runtime/south/plugins/transports/calvinfcm/twisted/twisted_transport.py

```diff
@@ -0,0 +1,278 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2015 Ericsson AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import inspect
+import time
+import traceback
+import base64
+import pdb
+
+from calvin.utilities.calvin_callback import CalvinCB, CalvinCBClass
+from calvin.utilities import calvinlogger
+from calvin.runtime.south.plugins.transports.lib.twisted import base_transport
+from calvin.runtime.south.plugins.transports import base_transport as calvin_base_transport 
+from calvin.utilities import calvinconfig
+from twisted.protocols.basic import Int32StringReceiver
+
+from twisted.internet import protocol, reactor, ssl
+from twisted.names.srvconnect import SRVConnector
+from twisted.words.protocols.jabber import client
+from twisted.words.xish import xmlstream as xishxmlstream
+from twisted.words.protocols.jabber import xmlstream
+
+from twisted.words.protocols.jabber.jid import JID
+
+_log = calvinlogger.get_logger(__name__)
+
+_CONF = calvinconfig.get()
+
+def create_uri(sender_id, token):
+    return "calvinfcm://%s:%s" % (sender_id, token)
+
+class CalvinXMLStreamProtocol(CalvinCBClass, xmlstream.XmlStream):
+
+    def __init__(self, *args, **kvargs):
+        self._authenticator = args[0]
+        self._callbacks = args[1]
+        self._uri = args[2]
+        self._server = args[3]
+        xmlstream.XmlStream.__init__(self, self._authenticator)
+        CalvinCBClass.__init__(self, self._callbacks)
+        self.counter = 0
+        #self._callback_execute('set_proto', self)
```

Comment:
I think it is. I must call the super constructors. The counter variable is used in the code to generate the FCM message id.